src/Text/Pandoc/Readers/HTML.hs

```diff
@@ -588,17 +588,23 @@ pFigure = try $ do
   skipMany pBlank
   let pImg  = (\x -> (Just x, Nothing)) <$>
                (pOptInTag "p" pImage <* skipMany pBlank)
-      pCapt = (\x -> (Nothing, Just x)) <$>
-               (pInTags "figcaption" inline <* skipMany pBlank)
+      pCapt = (\x -> (Nothing, Just x)) <$> do
+                skipMany pBlank
+                bs <- pInTags "figcaption" block
+                skipMany pBlank
+                let getInlines (Plain xs) = xs
+                    getInlines (Para xs)  = xs
+                    getInlines _          = mempty
+                return $ query getInlines bs
       pSkip = (Nothing, Nothing) <$ pSatisfy (not . matchTagClose "figure")
   res <- many (pImg <|> pCapt <|> pSkip)
   let mbimg = msum $ map fst res
   let mbcap = msum $ map snd res
   TagClose _ <- pSatisfy (matchTagClose "figure")
-  let caption = fromMaybe mempty mbcap
   case B.toList <$> mbimg of
-       Just [Image attr _ (url, tit)] ->
-         return $ B.para $ B.imageWith attr url ("fig:" ++ tit) caption
+       Just [Image attr alt (url, tit)] ->
+         let caption = fromMaybe alt mbcap
+         in  return $ B.para $ B.imageWith attr url ("fig:" ++ tit) $ B.fromList caption
```

Comment:
As noted in the linked issue, I'm not convinced we should do this.  The HTML doesn't have a caption, so why should we produce an image with a caption?  The tricky thing is that we're overloading `alt`; it will still be used for the image's alt text, which just disappears in the current setup.  I'm not for the alt attribute disappearing, but I'm also hesitant about creating a caption where there wasn't one before.